src/creatures/npcs/spawns/spawn_npc.cpp

```diff
@@ -77,8 +77,22 @@ bool SpawnsNpc::loadFromXml(const std::string& filenpcname)
 
 		for (auto childNode : spawnNode.children()) {
 			if (strcasecmp(childNode.name(), "npc") == 0) {
+				// If tag npc not have tag: name, then not load npc
 				pugi::xml_attribute nameAttribute = childNode.attribute("name");
 				if (!nameAttribute) {
+					SPDLOG_WARN("[SpawnsNpc::loadFromXml] - the npc tag is missing tag 'name' or is wrong");
+					continue;
+				}
+
+				// If tag npc not have tag: type, then not load npc
+				pugi::xml_attribute typeAttribute = childNode.attribute("type");
+				if (!typeAttribute) {
+					SPDLOG_WARN("[SpawnsNpc::loadFromXml] - Npc with name {} is missing tag 'type', is necessary to add a type (xml or lua)", nameAttribute.as_string());
+					continue;
+				}
+				// Not load npc if have the "xml" tag
+				std::string xmlType = childNode.attribute("type").as_string("xml");
+				if (boost::iequals(xmlType, "xml")) {
```

Comment:
I believe you don't need to load the attribute again, since it's stored in `typeAttribute ` already.
You could just do:
```
if (boost::iequals(typeAttribute.as_string(), "xml)) {
  continue;
}
```